### PR TITLE
add self keymap to lily58

### PR DESCRIPTION
**Pull request by gshmu**
share self keymap

### CODE REVIEW

File: keyboards/lily58/keymaps/gshmu/config.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (drashna) [inline keyboards/lily58/keymaps/gshmu/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/lily58/keymaps/gshmu/config.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#pragma once
+
+//#define USE_MATRIX_I2C
+
+/* Select hand configuration */
+
+#define MASTER_LEFT
+// #define MASTER_RIGHT
+// #define EE_HANDS
```
Comment (drashna) [inline keyboards/lily58/keymaps/gshmu/config.h:10]: default is master left, so no define is technically needed.

File: keyboards/lily58/keymaps/gshmu/config.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+#define USE_SERIAL_PD2
```
Comment (drashna) [inline keyboards/lily58/keymaps/gshmu/config.h:12]: deprecated. 
```suggestion
```

File: keyboards/lily58/keymaps/gshmu/config.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#define TAPPING_FORCE_HOLD
+#define PERMISSIVE_HOLD
+
+#define DYNAMIC_TAPPING_TERM_INCREMENT 10
+
+// Underglow
+/*
+#undef RGBLED_NUM
+#define RGBLED_NUM 14    // Number of LEDs
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/lily58/keymaps/gshmu/config.h:24]: depreciated.
```suggestion
```

File: keyboards/lily58/keymaps/gshmu/keymap.c [hunk 5]
```diff
@@ -0,0 +1,214 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/lily58/keymaps/gshmu/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/lily58/keymaps/gshmu/keymap.c [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+    '#', ';', '\'', '`', ',', '.', '/', ' ', ' ', ' '};
+
+void set_keylog(uint16_t keycode, keyrecord_t *record) {
+  char name = ' ';
+  if (keycode < 60) {
+    name = code_to_name[keycode];
+  }
+
+  // update keylog
+  snprintf(keylog_str, sizeof(keylog_str), "%dx%d, k%2d : %c",
```
Comment (drashna) [inline keyboards/lily58/keymaps/gshmu/keymap.c:148]: just a heads up, snprintf/sprintf can take up to ~1.5kb of firmware space. 

If you're okay ith that, that's okay, but you may want to check out:
https://docs.qmk.fm/#/squeezing_avr?id=oled-tweaks

As this is much more space friendly.

File: keyboards/lily58/keymaps/gshmu/rules.mk [hunk 7]
```diff
@@ -0,0 +1,6 @@
+VIA_ENABLE = yes
+OLED_ENABLE = no
```
Comment (drashna) [inline keyboards/lily58/keymaps/gshmu/rules.mk:2]: if you're not using the oled feature, you may want to just remove the oled code from your keymap, completely.

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by gshmu** [comment]
@drashna hello, I add gnu license and remove all unessery code

**Review by drashna — APPROVED** [review]

File: keyboards/lily58/keymaps/gshmu/keymap.c [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include <stdio.h>
+
```
Comment (fauxpark) [inline keyboards/lily58/keymaps/gshmu/keymap.c:19]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/lily58/keymaps/gshmu/keymap.c (hunk 8 above)
Comment (gshmu) [reply keyboards/lily58/keymaps/gshmu/keymap.c]: removed

**Review by fauxpark — APPROVED** [review]

**Pull request merged**